source/common/termination_predicate_impl.h

```diff
@@ -15,22 +17,34 @@ class TerminationPredicateBaseImpl : public TerminationPredicate {
     linked_child_ = std::move(child);
     return *linked_child_;
   }
+  TerminationPredicate& appendToChain(TerminationPredicatePtr&& child) final {
+    RELEASE_ASSERT(child != nullptr, "child == nullptr");
```

Comment:
I am wondering if this RELEASE_ASSERT follows the advice of the Envoy style guide:

https://github.com/envoyproxy/envoy/blob/master/STYLE.md

Wouldn't this crash in an obvious way if left without the assertion?